### PR TITLE
sinon: add fake server options

### LINKED ISSUE #21863: sinon.fakeServer.create(config?: FakeServerConfig)
```js
import * as sinon from 'sinon';
```

```js
sinon.fakeServer.create({
  autoRespond: true
})
```

`TS2554: Expected 0 arguments, but got 1.`

Definitions:

```js
interface SinonFakeServerStatic {
  create(): SinonFakeServer;
}
```
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/sinon/index.d.ts#L326

Sinon API:
http://sinonjs.org/releases/v2.1.0/fake-xhr-and-server
```js
// Creates a new server.
var server = sinon.fakeServer.create([config]);
```

Authors:
  - @mrbigdog2u
  - @rationull
  - @lumaxis
  - @nicojs
  - @43081j

### PR DESCRIPTION
**Pull request by 43081j**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://sinonjs.org/releases/v2.1.0/fake-xhr-and-server/
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Fixes #21863.

### CODE REVIEW

**Review by lumaxis — APPROVED** [review]

File: types/sinon/index.d.ts [hunk 1]
```diff
@@ -322,8 +318,15 @@ declare namespace Sinon {
         restore(): void;
     }
 
+    interface SinonFakeServerOptions {
+        autoRespond: boolean;
```
Comment (lumaxis) [inline types/sinon/index.d.ts]: Should these actually be each individually optional too? I guess you don't need to provide all options when you pass in a config object.

**Comment by 43081j** [comment]
looks like the tslint version got bumped since last time so we're failing lint!

will look into it

**Review by lumaxis — COMMENTED** [review]
Actually, just had a question about this.

File: types/sinon/index.d.ts (hunk 1 above)
Comment (43081j) [reply types/sinon/index.d.ts]: ah you have a very good point. im having some build issues locally so somehow didn't notice this!

File: types/sinon/index.d.ts [hunk 2]
```diff
@@ -293,14 +291,10 @@ declare namespace Sinon {
         FakeXMLHttpRequest: SinonFakeXMLHttpRequest;
     }
 
-    interface SinonFakeServer {
+    interface SinonFakeServer extends SinonFakeServerOptions {
```
Comment (nicojs) [inline types/sinon/index.d.ts]: Does this mean that `SinonFakeServer` will have optional fields for autoRespond, autoRespondAfter,...? That doesn't sound good. Couldn't we use TypeScript's `Partial` mapped type? All fields on `SinonFakeServerOptions` should be required and the `options` parameter should be of type `Partial<SinonFakeServerOptions>`

**Review by nicojs — CHANGES_REQUESTED** [review]
Looka good, but please don't break the existing api, see my remark

File: types/sinon/index.d.ts (hunk 2 above)
Comment (43081j) [reply types/sinon/index.d.ts]: The original types were wrong. These properties are optional already. In both the config object and the server instance. If you read the source of sinon's fake server, you'll see only the ones you pass get set, thus the others are undefined (optional).

Comment (nicojs) [reply types/sinon/index.d.ts]: Ah ok, than it's fine, thanks for the clarification :+1:

**Comment by 43081j** [comment]
Seems everyone is happy, can we get this merged?

File: types/sinon/index.d.ts (hunk 2 above)
Comment (43081j) [reply types/sinon/index.d.ts]: @nicojs i think you need to approve the pr so the bot will mark it as mergeable. can you take a look?

**Comment by nicojs** [comment]
👍

File: types/sinon/index.d.ts (hunk 2 above)
Comment (nicojs) [reply types/sinon/index.d.ts]: how? I just reacted with a thumbs up 👍

**Review by nicojs — APPROVED** [review]
Looks good 👍

**Pull request merged**